Tests/Test_SDL.cpp

```diff
@@ -0,0 +1,219 @@
+ * dirtsand is free software: you can redistribute it and/or modify           *
+ * it under the terms of the GNU Affero General Public License as             *
+ * published by the Free Software Foundation, either version 3 of the         *
+ * License, or (at your option) any later version.                            *
+ *                                                                            *
+ * dirtsand is distributed in the hope that it will be useful,                *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the              *
+ * GNU Affero General Public License for more details.                        *
+ *                                                                            *
+ * You should have received a copy of the GNU Affero General Public License   *
+ * along with dirtsand.  If not, see <http://www.gnu.org/licenses/>.          *
+ ******************************************************************************/
+
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
+
+#include <catch2/catch.hpp>
+#include <string_theory/format>
+
+#include <unistd.h>
+
+#include "SDL/DescriptorDb.h"
+#include "SDL/StateInfo.h"
+
+static const ST::string s_SdlDescriptor = ST_LITERAL(R"(
+    STATEDESC Test
+    {
+        VERSION 1
+
+        VAR BOOL    bTestVar1[1]    DEFAULT=0
+        VAR BOOL    bTestVar2[1]    DEFAULT=1    DEFAULTOPTION=VAULT
+        VAR INT     iTestVar3[1]    DEFAULT=0    DEFAULTOPTION=VAULT
+        VAR INT     iTestVar4[1]    DEFAULT=100
+    }
+
+    STATEDESC Test
+    {
+        VERSION 2
+
+        VAR BOOL    bTestVar1[1]    DEFAULT=0
+        VAR BOOL    bTestVar2[1]    DEFAULT=1    DEFAULTOPTION=VAULT
+        VAR INT     iTestVar3[1]    DEFAULT=0    DEFAULTOPTION=VAULT
+        VAR INT     iTestVar4[1]    DEFAULT=100
+
+        VAR BYTE    bTestVar5[1]    DEFAULT=50
+    }
+
+    STATEDESC Barney
+    {
+        VERSION 1
+    }
+)");
+
+static bool LoadDescriptors()
+{
+    do {
+        const char kTemplate[] = "DirtSandSDLTestXXXXXX";
+        char tempDir[256];
+        strcpy(tempDir, kTemplate);
+
+        char* tempDirResult = mkdtemp(tempDir);
+        if (!tempDirResult) {
+            fprintf(stderr, "Failed to create a temporary directory for SDL.\n");
+            return false;
+        }
+
+        ST::string sdlFilePath = ST::format("{}/Test.sdl", tempDirResult);
+        FILE* f = fopen(sdlFilePath.c_str(), "w+");
+        if (!f) {
+            fprintf(stderr, "Failed to create a file in our temporary SDL directory.\n");
+            return false;
+        }
+        fwrite(s_SdlDescriptor.c_str(), sizeof(char), s_SdlDescriptor.size(), f);
+        fclose(f);
+
+        SDL::DescriptorDb::LoadDescriptors(tempDir);
+        unlink(sdlFilePath.c_str());
+        rmdir(tempDirResult);
+        return true;
+    } while (0);
```

Comment:
I don't see any `break` or `continue`, and this is the only scope in the function...  Why is the `do ... while` necessary?

Reply:
Oof. I had the thought of using that construct for resource management and promptly failed to manage my resources correctly. 🤦‍♂️ Fixed now.